types-packages/code-actions.d.ts

```diff
@@ -13,7 +24,7 @@ export interface CodeActionProvider {
   getCodeActions(
     editor: Atom.TextEditor,
     range: Atom.Range,
-    diagnostics: Message[]
+    diagnostics: Diagnostic[] // | Message[]
```

Comment:
atom-languageclient uses Diagnostic. Not sure if linter Message supports code action API yet

Reply:
Diagnostic is from atom-ide-ui. Which is a slight variation on linter Message. Actually defined here: https://github.com/facebookarchive/atom-ide-ui/blob/master/modules/atom-ide-ui/pkg/atom-ide-diagnostics/lib/types.js#L93

I would suggest we rather scrub this legacy off of atom-languageclient rather than dragging it into the town square, figuratively speaking.

Since we're expecting Linter to handle our diagnostic messages, we really should use the Linter messages and not an ad-hoc approximation defined above.

Reply:
I agree. We should fix this issue. 

The `Message` type is not compatible with `Diagnostic`, and `Diagnostic` is what is used right now in atom-languageclient. Since it is part of a bigger change and codeAction is not implemented yet in the Linter package), I am going to keep `Diagnostic` now, so I can merge the linked PR.

The Diagnostic types were actually removed in this WIP PR, So, we should eventually do the big work, and fix everything.
https://github.com/atom-community/atom-languageclient/pull/112